httpRest/src/main/java/com/example/httpRest/controller/FacultyController.java

```diff
@@ -0,0 +1,49 @@
+package com.example.httpRest.controller;
+
+import com.example.httpRest.model.Faculty;
+import com.example.httpRest.service.FacultyService;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collection;
+
+@RestController
+@RequestMapping("/faculty")
+public class FacultyController {
+
+
+    private final FacultyService facultyService;
+
+    public FacultyController(FacultyService facultyService) {
+        this.facultyService = facultyService;
+    }
+
+    @GetMapping
+    public Collection<Faculty> getAll() {
+        return facultyService.getAll();
+    }
+
+    @GetMapping("/{id}")
+    public Faculty getById(@PathVariable("id") Long id) {
+        return facultyService.getById(id);
+    }
+
+    @PostMapping
+    public Faculty create(@RequestBody Faculty faculty) {
+        return facultyService.create(faculty);
+    }
+
+    @GetMapping("/filtred")
+    public Collection<Faculty> getByColor(@RequestParam("color") String color) {
+        return facultyService.getByColor(color);
+    }
+
+    @PutMapping("/id")
```

Comment:
Нужно подправить путь
```suggestion
    @PutMapping("/{id}")
```